travertino/tests/test_style.py

```diff
@@ -189,176 +189,63 @@ def test_apply(StyleClass):
-
-    # Set the value to something new
-    # A dirty notification is set.
-    style.explicit_value = 20
-    assert style.explicit_value == 20
-    style.apply.assert_called_once_with("explicit_value")
-
-    # Clear the applicator mock
-    style.apply.reset_mock()
-
-    # Clear the property
-    del style.explicit_value
-    assert style.explicit_value == 0
-    style.apply.assert_called_once_with("explicit_value")
-
-
-@pytest.mark.parametrize("StyleClass", [Style, DeprecatedStyle])
-def test_property_with_explicit_none(StyleClass):
-    style = StyleClass()
-
-    # Default value is None
-    assert style.explicit_none is None
-    style.apply.assert_not_called()
-
-    # Modify the value
-    style.explicit_none = 10
-
-    assert style.explicit_none == 10
-    style.apply.assert_called_once_with("explicit_none")
-
-    # Clear the applicator mock
-    style.apply.reset_mock()
-
-    # Set the property to the same value.
-    # No dirty notification is sent
-    style.explicit_none = 10
-    assert style.explicit_none == 10
-    style.apply.assert_not_called()
-
-    # Set the property to something new
-    # A dirty notification is set.
-    style.explicit_none = 20
-    assert style.explicit_none == 20
-    style.apply.assert_called_once_with("explicit_none")
-
-    # Clear the applicator mock
-    style.apply.reset_mock()
-
-    # Clear the property
-    del style.explicit_none
-    assert style.explicit_none is None
-    style.apply.assert_called_once_with("explicit_none")
-
-
-@pytest.mark.parametrize("StyleClass", [Style, DeprecatedStyle])
-def test_property_with_implicit_default(StyleClass):
+def test_property(name, initial, StyleClass):
     style = StyleClass()
 
-    # Default value is None
-    assert style.implicit is None
+    # Default value
+    assert style[name] == initial
     style.apply.assert_not_called()
 
     # Modify the value
-    style.implicit = 10
+    style[name] = 10
 
-    assert style.implicit == 10
-    style.apply.assert_called_once_with("implicit")
+    assert style[name] == 10
+    style.apply.assert_called_once_with(name)
 
     # Clear the applicator mock
     style.apply.reset_mock()
 
     # Set the value to the same value.
     # No dirty notification is sent
-    style.implicit = 10
-    assert style.implicit == 10
+    style[name] = 10
+    assert style[name] == 10
     style.apply.assert_not_called()
 
     # Set the value to something new
     # A dirty notification is set.
-    style.implicit = 20
-    assert style.implicit == 20
-    style.apply.assert_called_once_with("implicit")
+    style[name] = 20
+    assert style[name] == 20
+    style.apply.assert_called_once_with(name)
 
     # Clear the applicator mock
     style.apply.reset_mock()
 
     # Clear the property
-    del style.implicit
-    assert style.implicit is None
-    style.apply.assert_called_once_with("implicit")
+    del style[name]
+    assert style[name] == initial
+    style.apply.assert_called_once_with(name)
```

Comment:
The `explicit_const` version of this test does a second deletion to verify it's a no-op; that wasn't happening for any of the other tests in the parameterized group, but it seems like a test that is worth preserving.